test/keymap.cpp

```diff
@@ -519,3 +518,32 @@ TEST_CASE("Regression test for macro configuration semicolon handling",
 		REQUIRE(macros[1].args == std::vector<std::string>({}));
 	}
 }
+
+TEST_CASE("It's not an error to have no operations before a semicolon in "
+	"a macro",
+	"[KeyMap]")
+{
+	KeyMap k(KM_NEWSBOAT);
+	k.handle_action("macro", "r ; ;; ; open");
```

Comment:
Seeing that `;;` is accepted, I would expect any number of consecutive semicolons to be accepted (not important at all, feel free to ignore).